Fix build failure with GCC 11:

------
[...]
/tmp/guix-build-opencolorio-1.1.1.drv-0/source/src/core/ImageDesc.cpp:60:51: error: ‘this’ pointer is null [-Werror=nonnull]
   60 |             os << "width=" << packedImg->getWidth() << ", ";
      |                                                   ^
/tmp/guix-build-opencolorio-1.1.1.drv-0/source/src/core/ImageDesc.cpp:274:10: note: in a call to non-static member function ‘long int OpenColorIO::v1::PackedImageDesc::getWidth() const’
  274 |     long PackedImageDesc::getWidth() const
      |          ^~~~~~~~~~~~~~~
/tmp/guix-build-opencolorio-1.1.1.drv-0/source/src/core/ImageDesc.cpp:61:53: error: ‘this’ pointer is null [-Werror=nonnull]
   61 |             os << "height=" << packedImg->getHeight() << ", ";
      |                                                     ^
/tmp/guix-build-opencolorio-1.1.1.drv-0/source/src/core/ImageDesc.cpp:279:10: note: in a call to non-static member function ‘long int OpenColorIO::v1::PackedImageDesc::getHeight() const’
  279 |     long PackedImageDesc::getHeight() const
      |          ^~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
[...]
------

Patch copied from Gentoo:

https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=7e726d850502018b6760da78dbd4a419603016b8

diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 1eb691b6..cff9bd83 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -23,8 +23,6 @@ if(WIN32)
     if("${CMAKE_BUILD_TYPE}" STREQUAL "Release")
         set(EXTERNAL_COMPILE_FLAGS "${EXTERNAL_COMPILE_FLAGS} /WX")
     endif()
-else()
-    set(EXTERNAL_COMPILE_FLAGS "${EXTERNAL_COMPILE_FLAGS} -Werror")
 endif()
 
 # SHARED
